### ISSUE TITLE
Fix request: false positive for AvoidModifyingObjectMapper with a factory

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#155]
When a Factory class is used properly, we get a false positive
Example:
```
public class ObjectMapperFactory {
    private final ObjectMapper objectmapper = new ObjectMapper();

    public static ObjectMapperFactory builder() {
        return new ObjectMapperFactory();
    }

    public ObjectMapper build() {
        return objectmapper;
    }

    public ObjectMapperFactory annotationIntrospector(AnnotationIntrospector annotationIntrospector) {
        objectmapper.setAnnotationIntrospector(annotationIntrospector); // good, should not be flagged
        return this;
    }
}
```
Condition is: if method returns this and there is one and only one method that returns an ObjectMapper.
